### PR TITLE
feat: daily vault backup

### PR DESCRIPTION
**Pull request by artsyjian**
The type of this PR is: **Feat**

This PR supports [PHIRE-407]

### Description

Backup Vault data to S3 daily. And refactoring.

- Add vault_snapshot script/cronjob which does the Vault backup.
- Update src/lib/vault.py:
  - Support IAM login.
  - Add take_snapshot method.
- Move sanitizers and validators out of src/lib/util.py lib as it's becoming too long.
- Introduce src/lib/export_backup.py which absorbs s3 upload logic and local file logic from the scripts.
- Split migrate_config_secrets script's VAULT_ADDR env var into host and port vars.
- And misc. improvements.

dev/staging/production env has been updated.

Merging requires the following PRs to be merged first:

- [Related IAM changes.](https://github.com/artsy/infrastructure/pull/641)
- [Related DNS and Kubernetes changes.](https://github.com/artsy/substance/pull/368)

[PHIRE-407]: https://artsyproduct.atlassian.net/browse/PHIRE-407?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/kubernetes_export/export.py [hunk 1]
```diff
@@ -49,8 +48,6 @@ def validate(s3, s3_bucket):
   ''' validate config obtained from env and command line '''
   if s3 and not s3_bucket:
     raise Exception("K8S_BACKUP_S3_BUCKET must be specified in the environment.")
-  if s3 and not is_artsy_s3_bucket(s3_bucket):
-    raise Exception(f"{s3_bucket} seems not an Artsy S3 bucket.")
```
Comment (artsyjian) [inline src/kubernetes_export/export.py:53]: validation moved to export_backup.py

File: src/lib/util.py [hunk 2]
```diff
@@ -126,3 +99,7 @@ def unquote(str1):
     logging.debug(f'string is quoted with {quote_char}, removing quotes')
     return str1[1:-1]
   return str1
+
+def url_host_port(hostname, port=443):
+  ''' return https://hostname:port '''
+  return f'https://{hostname}:{port}'
```
Comment (artsyjian) [inline src/lib/util.py:105]: enforces https inicidentally.

**Comment by artsyjian** [comment]
[Related IAM changes.](https://github.com/artsy/infrastructure/pull/641)

**Comment by artsyjian** [comment]
[Related DNS and Kubernetes changes.](https://github.com/artsy/substance/pull/368)

File: hokusai/production.yml [hunk 3]
```diff
@@ -403,3 +403,42 @@ spec:
...
                     values:
                     - background
+
+---
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: opstools-vault-snapshot
+spec:
+  schedule: "42 7 * * *"
```
Comment (joeyAghion) [inline hokusai/production.yml:413]: An excellent choice

**Review by joeyAghion — APPROVED** [review]

**Comment by artsyjian** [comment]
can be merged

**Pull request merged**